nslsii/temperature_controllers/lakeshore.py

```diff
@@ -0,0 +1,126 @@
+from collections import OrderedDict
+from ophyd import (Component, DynamicDeviceComponent, Device, EpicsSignal,
+                   EpicsSignalRO, PVPositioner)
+
+
+class Lakeshore336(Device):
+    '''An ``ophyd.Device`` class to be used for Lakeshore336 temp controllers.
+
+    This is a class for controlling Lakeshore model 336 temperature controllers
+    The defaults are set to include four sensors (A, B, C and D) and four
+    control loops (1,2,3 and 4) however this is customizable at initialization
+    time via the kwargs ``temperatures`` and ``controls`` respectively. To
+    instantiate the default settings use the command:
+    ``my_controller = Lakeshore336(PVprefix, name='my_controller')``
+
+    Parameters
+    ----------
+    *args: list
+        The arguments that are passed to ``ophyd.Device``.
+    temperatures: list
+        A list of temperature sensor 'channels' to instantiate, the default is
+        ['A','B','C','D'].
+    controls: list
+        A list of heater control 'channels' to instantiate, the default is
+        [1,2,3,4].
+    **kwargs: dict
+        The keyword arguments passed to ``ophyd.Device``.
+    '''
+
+    def _set_fields(fields, device, prefix, **kwargs):
+        '''A function that allows for the components to be dynamically set.
+
+        Parameters
+        ----------
+        fields: list
+            A list of field identifiers to include, for ``_Temperature``
+            components an example is ['A','B','C','D'] and for ``_Control``
+            components an example is [1,2,3,4].
+        device: object
```

Comment:
Similar to above, this is not technically wrong (`type` instances are a sub-class of `object`) but this suggests you want _instances_ of the device, not the class of the device.

Reply:
resolved